compiler/rustc_expand/src/expand.rs

```diff
@@ -1706,9 +1733,29 @@ impl<'a, 'b> InvocationCollector<'a, 'b> {
             return match self.take_first_attr(&mut node) {
                 Some((attr, pos, derives)) => match attr.name_or_empty() {
                     sym::cfg => {
-                        if self.expand_cfg_true(&mut node, attr, pos) {
+                        let (res, meta_item) = self.expand_cfg_true(&mut node, attr, pos);
+                        if res {
                             continue;
                         }
+                        debug!("Stripping out node");
+                        if let Some(meta_item) = meta_item {
+                            let module_data = &self.cx.current_expansion.module;
```

Comment:
This is a very coarse-grained module info that is used for loading out-of-line `mod foo;` items from files.
I think you could use `current_expansion.lint_node_id` for more precisely locating false `cfg`s.

Then in resolve `node_id_to_def_id` and `def_id_to_node_id` can be used for relating that `NodeId` to `DefId`s.
If the `NodeId` doesn't have a corresponding `DefId` then this `cfg` certainly cannot add any names to modules and can be dropped.
If the corresponding `DefId` is the same as `base_error.module` in resolve then you have a match.

If you need to use some new name resolver functionality from expand you can add methods to `trait ResolverExpand`.

Reply:
That worked very well and allowed me to get rid of the cursed path matching logic in `find_cfg_stripped`, thanks!